backend/src/test/java/pl/krkteam/battleships/room/holder/RoomHolderTest.java

```diff
@@ -0,0 +1,142 @@
+package pl.krkteam.battleships.room.holder;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import pl.krkteam.battleships.common.domain.player.Player;
+import pl.krkteam.battleships.room.holder.dto.create.room.CreateRoomWrongDTO;
+import pl.krkteam.battleships.room.holder.dto.join.result.JoinResultOkDTO;
+import pl.krkteam.battleships.room.holder.dto.join.result.JoinResultWrongDTO;
+import pl.krkteam.battleships.wait.opponent.dto.OpponentAbsentDTO;
+import pl.krkteam.battleships.wait.opponent.dto.OpponentPresentDTO;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertEquals;
+
+public class RoomHolderTest {
+
+    private RoomHolder roomHolder;
+
+    @BeforeMethod
+    public void setUp() {
+        roomHolder = new RoomHolder();
+    }
+
+    @Test
+    public void testGetRoomListAndExpectHasTheSameRoomsNames() {
+        Player playerA = new Player("SomePlayerA");
+        Player playerB = new Player("SomePlayerB");
+        List<String> roomNamesList = Arrays.asList("SomeRoom1", "SomeRoom2");
+        roomHolder.createRoom(playerA, roomNamesList.get(0));
+        roomHolder.createRoom(playerB, roomNamesList.get(1));
```

Comment:
Code repeat in many methods below

+        Player playerA = new Player("SomePlayerA");
+        Player playerB = new Player("SomePlayerB");
+        List<String> roomNamesList = Arrays.asList("SomeRoom1", "SomeRoom2");
+        roomHolder.createRoom(playerA, roomNamesList.get(0));
+        roomHolder.createRoom(playerB, roomNamesList.get(1));

Reply:
It doesn't repeat in exactly the same same sequence, so it doesn't make any sense to  extract it to data provider